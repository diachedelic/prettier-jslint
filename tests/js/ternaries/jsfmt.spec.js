run_spec(__dirname, ["babel", "flow", "typescript"], { tabWidth: 4 });
