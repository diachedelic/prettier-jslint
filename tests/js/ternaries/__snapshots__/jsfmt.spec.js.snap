// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`binary.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
room = room.map((row, rowIndex) => (
  row.map((col, colIndex) => (
    (rowIndex === 0 || colIndex === 0 || rowIndex === height || colIndex === width) ? 1 : 0
  ))
))

=====================================output=====================================
room = room.map((row, rowIndex) => row.map((col, colIndex) => (
    rowIndex === 0 ||
    colIndex === 0 ||
    rowIndex === height ||
    colIndex === width
    ? 1
    : 0
)));

================================================================================
`;

exports[`func-call.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
fn(
  bifornCringerMoshedPerplexSawder,
  askTrovenaBeenaDependsRowans,
  glimseGlyphsHazardNoopsTieTie === averredBathersBoxroomBuggyNurl &&
    anodyneCondosMalateOverateRetinol
      ? annularCooeedSplicesWalksWayWay
      : kochabCooieGameOnOboleUnweave
);

fn(
  glimseGlyphsHazardNoopsTieTie
      ? annularCooeedSplicesWalksWayWay
      : kochabCooieGameOnOboleUnweave
);

(cancelAtPeriodEnd ? cancelSubscription : resumeSubscription)(
  req.user.email,
  req.route.key
)(callback);

=====================================output=====================================
fn(
    bifornCringerMoshedPerplexSawder,
    askTrovenaBeenaDependsRowans,
    (
        glimseGlyphsHazardNoopsTieTie === averredBathersBoxroomBuggyNurl &&
            anodyneCondosMalateOverateRetinol
        ? annularCooeedSplicesWalksWayWay
        : kochabCooieGameOnOboleUnweave
    )
);

fn(
    glimseGlyphsHazardNoopsTieTie
    ? annularCooeedSplicesWalksWayWay
    : kochabCooieGameOnOboleUnweave
);

(
    cancelAtPeriodEnd
    ? cancelSubscription
    : resumeSubscription
)(
    req.user.email,
    req.route.key
)(callback);

================================================================================
`;

exports[`indent.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
aaaaaaaaaaaaaaa ? bbbbbbbbbbbbbbbbbb : ccccccccccccccc ? ddddddddddddddd : eeeeeeeeeeeeeee ? fffffffffffffff : gggggggggggggggg

aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
?
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
?
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
?
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
:
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
:
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
:
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa

a
    ? {
        a: 0
      }
    : {
        a: {
             a: 0
           }
            ? {
                a: 0
              }
            : {
                y: {
                    a: 0
                }
                    ? {
                        a: 0
                    }
                    : {
                        a: 0
                    }
            }
      }

a
	? {
			a: function() {
				return a
					? {
							a: [
								a
									? {
											a: 0,
											b: [
												a
													? [
															0,
															1
													  ]
													: []
											]
									  }
									: [
											[
												0,
												{
													a: 0
												},
												a
													? 0
													: 1
											],
											function() {
												return a
													? {
															a: 0
													  }
													: [
															{
																a: 0
															},
															{}
													  ];
											}
									  ]
							]
					  }
					: [
							a
								? function() {
										a
											? a(
													a
														? {
																a: a(
																	{
																		a: 0
																	}
																)
														  }
														: [
																0,
																a(),
																a(
																	a(),
																	{
																		a: 0
																	},
																	a
																		? a()
																		: a(
																				{
																					a: 0
																				}
																		  )
																),
																a()
																	? {
																			a: a(),
																			b: []
																	  }
																	: {}
														  ]
											  ):
										a(
											a()
												? {
														a: 0
												  }
												: (function(a) {
														return a()
															? [
																	{
																		a: 0,
																		b: a()
																	}
															  ]
															: a(
																	[
																		a
																			? {
																					a: 0
																			  }
																			: {},
																		{
																			a: 0
																		}
																	]
															  );
												  })(
														a
															? function(a) {
																	return function() {
																		return 0;
																	};
															  }
															: function(a) {
																	return function() {
																		return 1;
																	}
															  }
												  )
										);
								  }
								: function() {

								  }
					  ];
			}
	  }
    : a;

=====================================output=====================================
(
    aaaaaaaaaaaaaaa
    ? bbbbbbbbbbbbbbbbbb
    : (
        ccccccccccccccc
        ? ddddddddddddddd
        : (
            eeeeeeeeeeeeeee
            ? fffffffffffffff
            : gggggggggggggggg
        )
    )
);

(
    aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
    ? (
        aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
        ? (
            aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
            ? aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
            : aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
        )
        : aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
    )
    : aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
);

(
    a
    ? {
        a: 0
    }
    : {
        a: (
            {
                a: 0
            }
            ? {
                a: 0
            }
            : {
                y: (
                    {
                        a: 0
                    }
                    ? {
                        a: 0
                    }
                    : {
                        a: 0
                    }
                )
            }
        )
    }
);

(
    a
    ? {
        a: function () {
            return (
                a
                ? {
                    a: [
                        (
                            a
                            ? {
                                a: 0,
                                b: [
                                    (
                                        a
                                        ? [0, 1]
                                        : []
                                    )
                                ]
                            }
                            : [
                                [
                                    0,
                                    {
                                        a: 0
                                    },
                                    (
                                        a
                                        ? 0
                                        : 1
                                    )
                                ],
                                function () {
                                    return (
                                        a
                                        ? {
                                            a: 0
                                        }
                                        : [
                                            {
                                                a: 0
                                            },
                                            {}
                                        ]
                                    );
                                }
                            ]
                        )
                    ]
                }
                : [
                    (
                        a
                        ? function () {
                            (
                                a
                                ? a(
                                    a
                                    ? {
                                        a: a({
                                            a: 0
                                        })
                                    }
                                    : [
                                        0,
                                        a(),
                                        a(
                                            a(),
                                            {
                                                a: 0
                                            },
                                            (
                                                a
                                                ? a()
                                                : a({
                                                    a: 0
                                                })
                                            )
                                        ),
                                        (
                                            a()
                                            ? {
                                                a: a(),
                                                b: []
                                            }
                                            : {}
                                        )
                                    ]
                                )
                                : a(
                                    a()
                                    ? {
                                        a: 0
                                    }
                                    : (function (a) {
                                        return (
                                            a()
                                            ? [
                                                {
                                                    a: 0,
                                                    b: a()
                                                }
                                            ]
                                            : a([
                                                (
                                                    a
                                                    ? {
                                                        a: 0
                                                    }
                                                    : {}
                                                ),
                                                {
                                                    a: 0
                                                }
                                            ])
                                        );
                                    })(
                                        a
                                        ? function (a) {
                                            return function () {
                                                return 0;
                                            };
                                        }
                                        : function (a) {
                                            return function () {
                                                return 1;
                                            };
                                        }
                                    )
                                )
                            );
                        }
                        : function () {}
                    )
                ]
            );
        }
    }
    : a
);

================================================================================
`;

exports[`nested.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
const message =
    i % 3 === 0 && i % 5 === 0 ?
        'fizzbuzz'
    : i % 3 === 0 ?
        'fizz'
    : i % 5 === 0 ?
        'buzz'
    :
        String(i)

const paymentMessage = state == 'success'
  ? 'Payment completed successfully'

: state == 'processing'
  ? 'Payment processing'

: state == 'invalid_cvc'
  ? 'There was an issue with your CVC number'

: state == 'invalid_expiry'
  ? 'Expiry must be sometime in the past.'

  : 'There was an issue with the payment.  Please contact support.'

const paymentMessage2 = state == 'success'
  ? 1 //'Payment completed successfully'

: state == 'processing'
  ? 2 //'Payment processing'

: state == 'invalid_cvc'
  ? 3 //'There was an issue with your CVC number'

: true //state == 'invalid_expiry'
  ? 4 //'Expiry must be sometime in the past.'

  : 5 // 'There was an issue with the payment.  Please contact support.'

=====================================output=====================================
const message =
    (
        i % 3 === 0 && i % 5 === 0
        ? "fizzbuzz"
        : (
            i % 3 === 0
            ? "fizz"
            : (
                i % 5 === 0
                ? "buzz"
                : String(i)
            )
        )
    );

const paymentMessage =
    (
        state == "success"
        ? "Payment completed successfully"
        : (
            state == "processing"
            ? "Payment processing"
            : (
                state == "invalid_cvc"
                ? "There was an issue with your CVC number"
                : (
                    state == "invalid_expiry"
                    ? "Expiry must be sometime in the past."
                    : "There was an issue with the payment.  Please contact support."
                )
            )
        )
    );

const paymentMessage2 =
    (
        state == "success"
        ? 1 //'Payment completed successfully'
        : (
            state == "processing"
            ? 2 //'Payment processing'
            : (
                state == "invalid_cvc"
                ? 3 //'There was an issue with your CVC number'
                : (
                    true //state == 'invalid_expiry'
                    ? 4 //'Expiry must be sometime in the past.'
                    : 5
                )
            )
        )
    ); // 'There was an issue with the payment.  Please contact support.'

================================================================================
`;

exports[`parenthesis.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
debug ? this.state.isVisible ? "partially visible" : "hidden" : null;
debug ? this.state.isVisible && somethingComplex ? "partially visible" : "hidden" : null;

a => a ? () => {a} : () => {a}
a => a ? a : a
a => a ? aasdasdasdasdasdasdaaasdasdasdasdasdasdasdasdasdasdasdasdasdaaaaaa : a

=====================================output=====================================
(
    debug
    ? (
        this.state.isVisible
        ? "partially visible"
        : "hidden"
    )
    : null
);
(
    debug
    ? (
        this.state.isVisible && somethingComplex
        ? "partially visible"
        : "hidden"
    )
    : null
);

(a) => (
    a
    ? () => {
        a;
    }
    : () => {
        a;
    }
);
(a) => (
    a
    ? a
    : a
);
(a) => (
    a
    ? aasdasdasdasdasdasdaaasdasdasdasdasdasdasdasdasdasdasdasdasdaaaaaa
    : a
);

================================================================================
`;
